src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -3163,15 +3236,86 @@ select string.IsNullOrEmpty(function.Namespace)
         {
             if (!typeLoadData.IsDSCompatible || typeLoadData.IsDeprecated || typeLoadData.IsHidden
                 || typeLoadData.IsMetaNode)
             {
-                return;
+                return null;
             }
-
+            var node = new NodeModelSearchElement(typeLoadData);
             SearchModel?.Add(new NodeModelSearchElement(typeLoadData));
+            return node;
+        }
+
+        /// <summary>
+        /// Initialize Lucene index document object for reuse
+        /// </summary>
+        /// <returns></returns>
+        private Document InitializeIndexDocument()
+        {
+            //TODO: all this harcoded string values should be moved to a different class.
+            var fullCategory = new TextField("FullCategoryName", "", Field.Store.YES);
+            var name = new TextField("Name", "", Field.Store.YES);
+            var description = new TextField("Description", "", Field.Store.YES);
+            var keywords = new TextField("SearchKeywords", "", Field.Store.YES);
+            var inp = new TextField("InputParameters", "", Field.Store.YES);
+            var outp = new TextField("OutputParameters", "", Field.Store.YES);
+
+            var docName = new StringField("DocName", "", Field.Store.YES);
+            var fullDoc = new TextField("Documentation", "", Field.Store.YES);
+
+            var pkgName = new TextField("PackageName", "", Field.Store.YES);
+            var pkgVer = new TextField("PackageVersion", "", Field.Store.YES);
+
+            var d = new Document()
+            {
+                fullCategory, name, description, keywords, inp,outp,
+                fullDoc, docName,
+                pkgName, pkgVer
+            };
+            return d;
+        }
+
+        /// <summary>
+        /// Add node information to Lucene index
+        /// </summary>
+        /// <param name="node">node info that will be indexed</param>
+        /// <param name="doc">Lucene document in which the node info will be indexed</param>
+        private void AddNodeTypeToSearchIndex(NodeSearchElement node, Document doc)
+        {
+            if (addedFields == null) return;
+
+            SetDocumentFieldValue(doc, "FullCategoryName", node.FullCategoryName);
+            SetDocumentFieldValue(doc, "Name", node.Name);
+            SetDocumentFieldValue(doc, "Description", node.Description);
+            SetDocumentFieldValue(doc, "InputParameters", string.Join(" ", node.InputParameters.Select(t => $"{t.Item1}:{t.Item2}")));
+            SetDocumentFieldValue(doc, "OutputParameters", node.OutputParameters.Aggregate((x, y) => x + " " + y));
+            if (node.SearchKeywords.Count > 0) SetDocumentFieldValue(doc, "SearchKeywords", node.SearchKeywords.Aggregate((x, y) => x + " " + y), true, true);
+
+            writer.AddDocument(doc);
+        }
+
+        private void SetDocumentFieldValue(Document doc, string field, string value, bool isTextField = true, bool isLast = false)
```

Comment:
Hey @RobertGlobant20, I wrote this function for the recharge, therefore I has the `isLast` option to quickly manage the remaining fields in the document, I am sure we can optimize or write this in a better way. I don't want to block the merge of this PR though, we can optimize it later, so can you leave a comment for this function? Specifically covering the last two parameters: `isTextField` and `isLast`
`isTextField` is used when the value need to be tokenized (broken down into pieces), whereas StringTextFields are tokenized.
`isLast` is used for the last value set in the document, and it will fetch all the other field not set for the document and add them with an empty string.

Reply:
Added the comments above the SetDocumentFieldValue method
commit: https://github.com/DynamoDS/Dynamo/pull/14013/commits/b5994fdcc8ef914d70dd347cfbe218babe0a5cb8